pkg/restapi/operation/operations.go

```diff
@@ -457,3 +460,80 @@ func (vc *VaultCollection) queryVault(vaultID string, query *models.Query) ([]st
 
 	return store.Query(query)
 }
+
+func validateDataVaultConfiguration(dataVaultConfig *models.DataVaultConfiguration) error {
+	if err := checkConfigRequiredFields(dataVaultConfig); err != nil {
+		return err
+	}
+
+	if err := edvutils.CheckIfURI(dataVaultConfig.Controller); err != nil {
+		return fmt.Errorf(messages.InvalidControllerString, err)
+	}
+
+	if err := checkFieldsWithURIArray(&dataVaultConfig.Invoker, dataVaultConfig.Controller); err != nil {
+		return fmt.Errorf(messages.InvalidInvokerStringArray, err)
+	}
+
+	if err := checkFieldsWithURIArray(&dataVaultConfig.Delegator, dataVaultConfig.Controller); err != nil {
+		return fmt.Errorf(messages.InvalidDelegatorStringArray, err)
+	}
+
+	if err := edvutils.CheckIfURI(dataVaultConfig.KEK.ID); err != nil {
+		return fmt.Errorf(messages.InvalidKEKIDString, err)
+	}
+
+	return nil
+}
+
+func checkConfigRequiredFields(config *models.DataVaultConfiguration) error {
+	if config.Controller == "" {
+		return errors.New(messages.BlankController)
+	}
+
+	if config.KEK.ID == "" {
+		return errors.New(messages.BlankKEKID)
+	}
+
+	if config.KEK.Type == "" {
+		return errors.New(messages.BlankKEKType)
+	}
+
+	if config.HMAC.ID == "" {
+		return errors.New(messages.BlankHMACID)
+	}
+
+	if config.HMAC.Type == "" {
+		return errors.New(messages.BlankHMACType)
+	}
+
+	return nil
+}
+
+// If the given array is empty, populate it with the default value. If it's not empty, check if every string in the
+// array is a valid URI.
+func checkFieldsWithURIArray(arr *[]string, defaultVal string) error {
+	if len(*arr) == 0 {
+		*arr = make([]string, 1)
+		(*arr)[0] = defaultVal
+
+		return nil
+	}
+
+	if err := edvutils.CheckIfArrayIsURI(*arr); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func validateIncomingEncryptedDocument(doc models.EncryptedDocument) error {
+	if encodingErr := edvutils.CheckIfBase58Encoded128BitValue(doc.ID); encodingErr != nil {
+		return encodingErr
+	}
+
+	if doc.JWE == nil {
+		return errors.New(messages.BlankJWE)
+	}
```

Comment:
In addition to checking that it's not nil, you should also make sure it's not just an empty `json.RawMessage`.

I think there's something else we should check in the JWE at a minimum... @Baha-sk, @troyronda, what else in the JWE should we check? I _think_ someone told me once that we should check the "alg" field?

Reply:
@DRK3 

> In addition to checking that it's not nil, you should also make sure it's not just an empty json.RawMessage.

`json.RawMessage` is a typed `[]byte`. It's enough to just check for length.

> I think there's something else we should check in the JWE at a minimum... @Baha-sk, @troyronda, what else in the JWE should we check? I think someone told me once that we should check the "alg" field?

Must ban `none`.